### PR TITLE
Bot.js - Support for Chaining Methods #10

### PR DESCRIPTION
**Pull request by marekhakala**
Hi, I prepared a patch for Chaining Methods of Bot class. Please review my code and write down your feedback. Thx #10

### CODE REVIEW

**Comment by manthanhd** [comment]
Hey can you please add some tests? Also you missed the resolve method.

**Comment by manthanhd** [comment]
Nice work BTW! :D

**Comment by manthanhd** [comment]
Also I'd love if you could mention the chain ability in the documentation too!

File: test/BotTest.js [hunk 1]
```diff
@@ -618,4 +618,52 @@ describe('Bot', function () {
...
+    describe('ChainableMethods', function () {
+        it('Chainable methods with train and addSkill', function (done) {
+            var mockClassifier = mockClassifierWithMockClassifierFactory();
+
+            var bot = new Bot({classifierPreference: 'naive_bayes'});
+
+            const fakeSkillFn = function (context, req, res, next) {};
+            const fakeSkill = new Skill('fakeskill', 'topic', fakeSkillFn);
+
+            const skills = bot.train('topic', 'text').addSkill(fakeSkill).getSkills();
```
Comment (manthanhd) [inline test/BotTest.js]: I see what you are trying to do here, however, I think this here is more of an integration test than a unit test. A unit test in this case would be something like:
```javascript
const bot = new Bot();
const returnReference = bot.addSkill(new Skill(...));
expect(returnReference).toBe(bot);
```

You'd have multiple tests for each of the methods that you have changed, testing multiple flows, ensuring that you get the reference back via the right flows.

Unit test needs to be small and concise; something which tests precisely one unit.

What do you think?

File: README.md [hunk 2]
```diff
@@ -154,6 +154,13 @@ When you run your code, you should get two messages back:
     content: 'You asked: "How\'s it going? Assistance required please.". I can tell you how I\'m doing if you ask nicely.' } ]
 ```
 
+### Chainable methods
+Currently `train`, `addSkill` and `resolve` methods support chainable attitude. That's mean you can create Bot object and cascade methods like is mentioned below.
```
Comment (manthanhd) [inline README.md]: Minor grammar...
Change:
```
That's mean
```

to:
```
That means
```

Comment (manthanhd) [inline README.md]: Minor grammar...
Change:
```
... support chainable attitude.
```

to:
```
are chainable.
```

File: test/BotTest.js [hunk 3]
```diff
@@ -618,4 +618,52 @@ describe('Bot', function () {
             expect(classifier).toBe(fakeClassifier);
         });
     });
-});
\ No newline at end of file
+
+    describe('ChainableMethods', function () {
+        it('Chainable methods with train and addSkill', function (done) {
+            var mockClassifier = mockClassifierWithMockClassifierFactory();
```
Comment (manthanhd) [inline test/BotTest.js]: You shouldn't need the mockClassifier here if you are not going to use it. The classifier is only used when you call the resolve method which, from what I can see is not called in this test.

Comment (marekhakala) [inline test/BotTest.js]: Ok, I remove mockClassifier variable from next PR.

File: test/BotTest.js (hunk 1 above)
Comment (marekhakala) [inline test/BotTest.js]: Sure, I prepare new tests that will be based on your example. That makes sense.

File: README.md (hunk 2 above)
Comment (marekhakala) [inline README.md]: OK

Comment (marekhakala) [inline README.md]: OK

**Comment by marekhakala** [comment]
Ok, I have it done. Check the PR now, please.

**Comment by manthanhd** [comment]
Thank you @marekhakala for your contribution 😃

**Pull request merged**